.travis.yml

```diff
@@ -8,10 +8,7 @@ addons:
       - oracle-java9-installer
 
 before_install:
-  - wget https://archive.apache.org/dist/maven/maven-3/3.5.0/binaries/apache-maven-3.5.0-bin.zip
-  - unzip -qq apache-maven-3.5.0-bin.zip
-  - export M2_HOME=$PWD/apache-maven-3.5.0
-  - export PATH=$M2_HOME/bin:$PATH
+  - mvn -N io.takari:maven:wrapper
```

Comment:
I don't feel like going through a CQ is worth it, so I removed the Apache code and binaries. Instead, I'm now treating the maven-wrapper maven plugin just like an improved maven downloader, and replacing the wget stuff with it.

The advantages are:
* It automatically downloads the latest maven. We can lock it to a single version if we want, but I don't particularly want to. Our builds should be able to run on pretty much any maven 3.x.
* It downloads maven jars into .m2/ which gets cached. And the wrapper plugin will also get cached.
* This will bump us to maven 3.5.4 which has improved logs, which I do actually want in our travis builds.